Use WebpackDevServer

fiskus opened the pull request.

nl0 commented on catalog/internals/webpack/webpack.prod.js:
```diff
@@ -6,7 +6,6 @@ const TerserPlugin = require('terser-webpack-plugin')
   mode: 'production',
 
-  // In production, we skip all hot-reloading stuff
   entry: {
```
if this is the same in both prod and dev, we can move it to the base

fiskus replied:
It's different for desktop

nl0 commented on catalog/package.json:
```diff
@@ -20,12 +20,9 @@
-    "start": "cross-env NODE_ENV=development node server",
-    "start:tunnel": "cross-env NODE_ENV=development ENABLE_TUNNEL=true node server",
-    "start:production": "npm run test && npm run build && npm run start:prod",
-    "start:prod": "cross-env NODE_ENV=production node server",
```
so, we don't have a way to start the node server in production mode now? i know it's not used in production deployments, but i was using it to test the production build

nl0 commented on catalog/package.json:
```diff
@@ -20,12 +20,9 @@
-    "presetup": "npm i chalk shelljs",
-    "setup": "node ./internals/scripts/setup.js",
+    "build:desktop": "cross-env NODE_ENV=production webpack --config internals/webpack/webpack.desktop-prod.js --color --progress",
+    "desktop": "cross-env NODE_ENV=development webpack serve --config internals/webpack/webpack.desktop-dev.js",
```
these files (desktop-prod and desktop-dev) are missing from the PR

nl0 commented on catalog/package.json:
```diff
@@ -67,16 +64,14 @@
     "dompurify": "^2.3.3",
     "echarts": "^5.2.2",
-    "express": "^4.17.1",
+    "electron": "^13.1.7",
```
why do we need electron in the webapp repo?

nl0 commented on catalog/server/middlewares/frontendMiddleware.js:
```diff
@@ -1,66 +0,0 @@
-    })
-  }
-
-  app.get('/__embed-debug', sendFile('embed-debug-harness.html'))
```
does devserver support these routes somehow?

nl0 reviewed (commented):
well, the first question is why do we need this?
the second is are we dropping our custom routes, e.g. `/__embed` etc?

fiskus commented:
> why do we need this?
1. It's easier to maintain
2. Live reload is fixed

fiskus commented:
* Made `/__embed-debug` route works using `devServer.historyApiFallback.rewrites`
* Reverted `./server` and `npm run start:prod` removal
* Removed extra dependencies, that will be introduced later but not now

nl0 reviewed (approved):
looking good now
